File: Greedy/452.Minimum-Number-of-Arrows-to-Burst-Balloons/452.Minimum-Number-of-Arrows-to-Burst-Balloons.cpp
class Solution {
    static bool cmp(vector<int>&a, vector<int>&b)
    {
        return a[1] < b[1];
    }
    
public:
    int findMinArrowShots(vector<vector<int>>& points) 
    {
        sort(points.begin(), points.end(), cmp);
        int i = 0;
        int count = 0;
        while (i<points.size())
        {
            count++;
            int j = i+1;
            while (j<points.size() && points[j][0]<=points[i][1])
                j++;
            i = j;
        }
        return count;        
    }
};

File: Sort/452.Minimum-Number-of-Arrows-to-Burst-Balloons/452.Minimum-Number-of-Arrows-to-Burst-Balloons.py
class Solution:
    def findMinArrowShots(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        points = sorted(points, key=lambda x:x[1])
        j = 0
        count = 0
        while (j<len(points)) :
            right = points[j][1]
            while (j<len(points) and points[j][0]<=right):
                j+=1
            count+=1
        return count
    
